src/components/store/AnalyzeStore.ts

```diff
@@ -4,8 +4,33 @@ import { analyzeDataType } from '@/types/type'
 
 interface AnalyzeStoreType {
 	consumptionData: analyzeDataType[]
+	consumptionCombinedData: analyzeDataType[]
+}
+
+//카테고리대로 가격 더하기
+function combiningData(consumptionData: analyzeDataType[]) {
+	const result = Object.values(
+		consumptionData.reduce(
+			(acc, item) => {
+				if (!acc[item.category]) {
+					acc[item.category] = { ...item }
+				} else {
+					acc[item.category].price += item.price
+				}
+				return acc
+			},
+			{} as Record<string, (typeof consumptionMockData)[number]>,
```

Comment:
`as` 타입 단언을 사용하는 것보다 reduce에 타입 추론을 명시해주는 게 좋을 것 같습니다.
```suggestion
	const result = Object.values(
		consumptionData.reduce<Record<string, analyzeDataType>(
		    // 여기서 acc는 Record<string, analyzeDataType> 타입
                   // item은 analyzeDataType 타입
			(acc, item) => {
				if (!acc[item.category]) {
					acc[item.category] = { ...item }
				} else {
					acc[item.category].price += item.price
				}
				return acc
			},
			{}
```

Reply:
as 타입 선언을 한번도 써본적이 없어서 한번 써봤습니다.. ㅎㅎ
찾아보니 안티패턴의 대표 사례네요
[쓰지말라는 Type Assertions(타입 단언) 은 왜 있나?](https://velog.io/@kmh060020/%EC%93%B0%EC%A7%80%EB%A7%90%EB%9D%BC%EB%8A%94-Type-Assertions%ED%83%80%EC%9E%85-%EB%8B%A8%EC%96%B8-%EC%9D%80-%EC%99%9C-%EC%9E%88%EB%82%98)
1. HTML 요소 접근
2. parsing된 JSON 데이터 타입 선언
정도에만 쓴다고 하는군요, 수정하겠습니다~